src/component/modal/editRange/forms/components/SignalsForm.jsx

```diff
@@ -0,0 +1,221 @@
+import { jsx, css } from '@emotion/core';
+/** @jsx jsx */
+import { useFormikContext } from 'formik';
+import { useCallback, useMemo, memo, useEffect, useState } from 'react';
+
+import { useChartData } from '../../../../context/ChartContext';
+import { Tabs } from '../../../../elements/Tab';
+import { translateMultiplet } from '../../../../panels/extra/utilities/MultiplicityUtilities';
+import Events from '../../../../utility/Events';
+
+import AddSignalFormTab from './AddSignalFormTab';
+import SignalFormTab from './SignalFormTab';
+
+const textStyles = css`
+  text-align: center;
+  width: 100%;
+
+  .errorText {
+    color: red;
+  }
+
+  .infoText {
+    margin-bottom: 20px;
+    font-size: 11px;
+  }
+`;
+
+const tabStylesAddition = css`
+  color: red;
+`;
+
+const SignalsForm = memo(() => {
+  const { values, setFieldValue, errors } = useFormikContext();
+
+  const { data: spectraData, activeSpectrum } = useChartData();
+  const [activeField, setActiveField] = useState(null);
+  const [frequency, setFrequency] = useState(null);
+
+  useEffect(() => {
+    if (
+      spectraData &&
+      activeSpectrum &&
+      spectraData[activeSpectrum.index] &&
+      spectraData[activeSpectrum.index].info &&
+      spectraData[activeSpectrum.index].info.originFrequency
+    ) {
+      setFrequency(spectraData[activeSpectrum.index].info.originFrequency);
+    } else {
+      setFrequency(null);
+    }
+  }, [activeSpectrum, spectraData]);
+
+  useEffect(() => {
+    const unsubscribe = Events.subscribe('brushEnd', (event) => {
+      if (values.activeTab !== 'addSignalTab' && activeField && frequency) {
+        setFieldValue(
+          activeField,
+          Math.abs(event.range[0] - event.range[1]) * frequency,
+        );
+      }
+    });
+
+    return () => {
+      unsubscribe();
+    };
+  }, [activeField, setFieldValue, frequency, values.activeTab]);
+
+  useEffect(() => {
+    const unsubscribe = Events.subscribe('mouseClick', (event) => {
+      if (values.activeTab === 'addSignalTab' && activeField) {
+        setFieldValue(activeField, event.xPPM);
+      }
+    });
+
+    return () => {
+      unsubscribe();
+    };
+  }, [values.activeTab, activeField, setFieldValue]);
+
+  const handleOnFocus = useCallback(
+    (name) => {
+      setActiveField(name);
+    },
+    [setActiveField],
+  );
+
+  const handleOnBlur = useCallback(() => {
+    setActiveField(null);
+  }, []);
+
+  const tapClickHandler = useCallback(
+    ({ tabid }) => {
+      setFieldValue('activeTab', tabid);
+    },
+    [setFieldValue],
+  );
+
+  const onDeleteSignal = useCallback(
```

Comment:
any handler you can name it in two way, deleteSignalHandler or handleDeleteSignal but not onDeleteSignal